navigation_and_routing/lib/src/data/book.dart

```diff
@@ -0,0 +1,15 @@
+// Copyright 2021, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'author.dart';
+
+class Book {
+  final int id;
+  final String title;
+  late final Author author;
+  final bool isPopular;
+  final bool isNew;
+
+  Book(this.id, this.title, this.isPopular, this.isNew);
```

Comment:
nit: consider making the boolean arguments named required for better readability on the call side.